source/fundamentals/linq.txt

```diff
@@ -320,6 +320,71 @@ following documents:
    { "date" : ISODate("2012-12-05T00:00:00Z"), "grade" : "A", "score" : 13 }
    { "date" : ISODate("2012-05-17T00:00:00Z"), "grade" : "A", "score" : 11 }
 
+Nested Statements
++++++++++++++++++
+
+You can chain or nest ``Select`` and ``SelectMany`` statements. The examples
+in this section use the ``Guidebook`` class, which contain an array of ``Restaurant``
+objects. The following code shows an example of a document modeled by the 
+``Guidebook`` class:
+
+.. code-block:: json
+
+   { 
+      "_id": { "$oid": ... },
+      "restaurants": [
+         {
+            "_id": { ... } ,
+            "address": { ... },
+            "name": "Tov Kosher Kitchen",
+            ...
+         },
+         {
+            "_id": { ... } ,
+            "address": { ... },
+            "name": "Harriet's Kitchen",
+            ...
+         },
+         ...
+      ]
+   }
+
+The following code models the ``Guidebook`` class:
+
+.. literalinclude:: /includes/fundamentals/code-examples/linq.cs
+   :language: csharp
+   :start-after: start-guidebook-model
+   :end-before: end-guidebook-model
+
+You can nest ``SelectMany`` statements within ``SelectMany`` or ``Select``
+statements. The following example nests a ``SelectMany`` statement within a
+``Select`` statement to retrieve an array of grades from all restaurants
+in a document modeled by the ``Guidebook`` class.
+
```

Comment:
I: since the examples on this page use the "sample_restaurants.restaurants" collection, which the "queryableCollection" variable represents, I think it gets confusing to introduce "Guidebooks". "Restaurants" kinda has to be the top-level class, since each instance represents a "restaurants" document. Is there a way to nest SelectMany within Select without introducing Guidebooks -- maybe selecting for restaurant documents with a certain "cuisine" value, then unwinding the grades arrays for just those documents?

Reply:
While I agree that it is confusing to introduce "Guidebooks," unfortunately filtering by value requires `Where` instead of `Select`. The ideal example for this would involve nested arrays within a document. I tried to update the example to be a bit less involved, but I'm not sure if it's better or worse.

Reply:
ah okay! Then I think it's fine to introduce a new sample document